### PR TITLE
Added new rules for command-line arguments

### PR DESCRIPTION
**Pull request by davkean**
Splitting up the new language service into management review-able chunks.

- Add two new rules for retrieving command-line arguments during design-time builds. These will be pulled used by the new language service host.
- Add new CSharp and VisualBasic project properties classes to hang these rules off.
- Copy rules to output directory to make it easier to test.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp/Microsoft.VisualStudio.ProjectSystem.CSharp.csproj [hunk 1]
```diff
@@ -35,10 +35,27 @@
...
   </ItemGroup>
+  <ItemGroup>
+    <XamlPropertyRule Include="ProjectSystem\Rules\CSharp.CscCommandLineArgs.xaml">
+      <Generator>MSBuild:GenerateRuleSourceFromXaml</Generator>
+      <SubType>Designer</SubType>
+    </XamlPropertyRule>
+  </ItemGroup>
   <Import Project="..\..\build\Targets\VSL.Imports.targets" />
+  <ItemDefinitionGroup>
+    <!-- Change the defaults for all rules so that they generate to C# properties -->
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp/Microsoft.VisualStudio.ProjectSystem.CSharp.csproj]: So even the properties in the BrowseObject rules are in CSharpProjectProperties? Shouldn't it be just the ones published to the Project context?

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp/Microsoft.VisualStudio.ProjectSystem.CSharp.csproj]: My understanding is that all properties apply to ProjectProperites, regardless of context.

**Comment by srivatsn** [comment]
👍

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp/ProjectSystem/Rules/CSharpProjectProperties.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /// </summary>
+    [Export]
+    [ExcludeFromCodeCoverage]
+    internal partial class CSharpProjectProperties : ProjectProperties
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProjectProperties"/> class.
+        /// </summary>
+        [ImportingConstructor]
+        public CSharpProjectProperties([Import] ConfiguredProject configuredProject)
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp/ProjectSystem/Rules/CSharpProjectProperties.cs]: If we can think of a better name here that'll be good. ProjectProperties is very confusing when you consider that CPS's IProjectPropertiesProvider is something else.

Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp/ProjectSystem/Rules/CSharpProjectProperties.cs]: How about ProjectRules? Writing ProjectRules.GetConfigurationGeneralAsync().GetTargetFrameworkAsync() seems reasonable.

**Pull request merged**